packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -126,7 +126,7 @@ function GridRow(props: React.HTMLAttributes<HTMLDivElement> & GridRowProps) {
   const publish = React.useCallback(
     (
         eventName: keyof GridRowEventLookup,
-        propHandler: React.MouseEventHandler<HTMLDivElement> | undefined,
+        propHandler?: React.MouseEventHandler<HTMLDivElement> | undefined,
```

Comment:
A click in the reorder handle shouldn't fire `rowClick`. You need to add a check for it in this file. The idea is:

`cellClick` is triggered for clicks everywhere
`rowClick` is not triggered on clicks in built-in columns with interactive elements